include/tvm/runtime/container.h

```diff
@@ -274,6 +276,137 @@ class ADT : public ObjectRef {
   TVM_DEFINE_OBJECT_REF_METHODS(ADT, ObjectRef, ADTObj);
 };
 
+/*! \brief An object representing string. It's POD type. */
+class StringObj : public Object {
+ public:
+  /*! \brief The length of the string object. */
+  uint32_t size;
+
+  /*! \brief The pointer to string data. */
```

Comment:
Need _type_key etc.

Regression coverage:
- Add testcase that assigns an String to ObjectRef, then downcast back to String